chanzuckerberg/czgenepi
# thuang-pangolin-127257

### Description

1. Implement Lineage feature per [Figma](https://www.figma.com/file/uHpPNuDSBupK97ORCkH1op/Aspen-v0-to-v2?node-id=114%3A4325)

#### Issue
https://app.clubhouse.io/genepi/story/127257/show-pangolin-lineages-for-each-sample-in-the-sampletableview

### Test plan

1. Visit: https://thuang-lineage-frontend.dev.genepi.czi.technology/ (UPDATE: RDEV BE is having DB migration issue, so please run the branch locally for now 🙏 Thank you!)
2. You should be able to see Lineage column now
3. Hover over the Lineage header, and the header font should turn black and show the tooltip. The tooltip should be interactive, which means the "Learn more" link is clickable, which will open in a new tab
4. Lineage cell on hover should show row highlight color, should also show tooltip with details, which is NOT interactive. This means that the content of the tooltip cannot be copied, since the tooltip will go away as soon as the cursor exits the Tooltip target (the version value in the cell). Also the cell on hover should show dotted underscore
5. In the first column, it should now show either "complete" or "failed" label. RIght now it seems like all the samples have "complete" labels. 

### Known issues:
1. Deviation from design is that the label is not in the same row as the text, since there's not enough width for that. We can fix this later if needed to always give the first column more width than other columns, we just need to decide how much extra width
2. Another one is that the Lineage detail tooltip currently has an arrow, which we don't have in Figma. This will be fixed when IDseq team updates the Tooltip component in the library. Or if it's not done, I'll do a quick patch to the Tooltip to allow arrow removal before this Friday

![demo](https://user-images.githubusercontent.com/6309723/117718883-9437ef00-b191-11eb-98ba-29b7d9b4af31.gif)
<img width="1065" alt="Screen Shot 2021-05-10 at 1 02 00 PM" src="https://user-images.githubusercontent.com/6309723/117718904-9d28c080-b191-11eb-87d2-b4c7ef13a636.png">
<img width="459" alt="Screen Shot 2021-05-10 at 1 02 11 PM" src="https://user-images.githubusercontent.com/6309723/117718907-9dc15700-b191-11eb-8012-b9853c82634c.png">
<img width="439" alt="Screen Shot 2021-05-10 at 1 02 14 PM" src="https://user-images.githubusercontent.com/6309723/117718909-9e59ed80-b191-11eb-86fb-9c69c20688d1.png">
